python/lsst/meas/extensions/multiprofit/consolidate_astropy_table.py

```diff
@@ -196,18 +210,32 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
 
         tables_read = []
         check_columns = self.config.drop_duplicate_columns or self.config.validate_duplicate_columns
+        n_bands = len(bands_sorted)
 
         for name, data_name in data.items():
             config_input = self.config.inputs[name]
-            tables = [
-                (
-                    apTab.hstack([data_name[patch][band] for band in bands_sorted], join_type="exact")
-                    if not config_input.is_multiband
-                    else data_name[patch][band_null]
-                )
-                for patch in (patches_ref if not config_input.is_multipatch else patches_null)
-            ]
-            table_new = tables[0] if (len(tables) == 1) else apTab.vstack(tables, join_type="exact")
+            tables = []
+            bands_missing = False
+            for patch in (patches_ref if not config_input.is_multipatch else patches_null):
+                data_name_patch = data_name[patch]
+                if config_input.is_multiband:
+                    tables_patch = data_name_patch.get(band_null, [])
+                else:
+                    tables_patch = [
+                        _tab for band in bands_sorted if (_tab := data_name_patch.get(band)) is not None
+                    ] if not config_input.is_multiband else data_name[patch][band_null]
+                    if not bands_missing and (len(tables_patch) != n_bands):
+                        bands_missing = True
+                # Join only if there's something to join
+                if tables_patch:
+                    table_patch = apTab.hstack(tables_patch, join_type="exact")
+                    tables.append(table_patch)
+                # If there's nothing to join, presumably the task failed
+                # stacking should handle some tasks failing but not others, but
+                # this shouldn't be relied upon
+
+            table_new = tables[0] if (len(tables) == 1) else apTab.vstack(
+                tables, join_type="outer" if bands_missing else "exact")
```

Comment:
vstack will vstack even if len==1, but any chance that there are no tables by the time we get here?

Reply:
There is, but this is looping over dataset types, so that would mean there are no datasets of that type at all in the entire tract and this is going to fail or at best return an empty table later anyway.